kafka/src/main/java/com/opentable/logging/PartitionKeyGenerator.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.opentable.logging;
+
+class PartitionKeyGenerator {
+    private final ThreadLocal<byte[]> partitionShufflerKey = ThreadLocal.withInitial(() -> new byte[8]);
+
+    byte[] next() {
+        final byte[] key = partitionShufflerKey.get();
+
+        for (int i = key.length-1; i >= 0; i--) {
+            if (key[i]++ != Byte.MIN_VALUE) {
+                break;
+            }
+        }
+
+        return key;
+    }
+}
```

Comment:
Why not just do a `ThreadLocal<Long>`, increment, and convert to a byte array for the key when passing back to Kafka?  An increment would be much easier to read and understand than this loop.